solc/CommandLineParser.cpp

```diff
@@ -0,0 +1,1239 @@
+			return false;
+		}
+
+		try
+		{
+			yul::OptimiserSuite::validateSequence(m_args[g_strYulOptimizations].as<string>());
+		}
+		catch (yul::OptimizerException const& _exception)
+		{
+			serr() << "Invalid optimizer step sequence in --" << g_strYulOptimizations << ": " << _exception.what() << endl;
+			return false;
+		}
+
+		m_options.yulOptimiserSteps = m_args[g_strYulOptimizations].as<string>();
+	}
+
+	if (m_args.count(g_argImportAst) > 0)
+		m_options.inputMode = InputMode::CompilerWithASTImport;
+	else
+		m_options.errorRecovery = (m_args.count(g_argErrorRecovery) > 0);
+
+	solAssert(m_options.inputMode == InputMode::Compiler || m_options.inputMode == InputMode::CompilerWithASTImport, "");
+	return true;
+}
+
+bool CommandLineParser::hasOutput()
+{
+	return g_hasOutput;
+}
+
+bool CommandLineParser::parseCombinedJsonOption()
+{
+	if (!m_args.count(g_argCombinedJson))
+		return true;
+
+	set<string> requests;
+	for (string const& item: boost::split(requests, m_args[g_argCombinedJson].as<string>(), boost::is_any_of(",")))
+		if (!g_combinedJsonArgs.count(item))
+		{
+			serr() << "Invalid option to --" << g_argCombinedJson << ": " << item << endl;
+			return false;
+		}
+
+	m_options.combinedJsonRequests = CombinedJsonRequests{};
+	m_options.combinedJsonRequests->abi = (requests.count(g_strAbi) > 0);
+	m_options.combinedJsonRequests->metadata = (requests.count("metadata") > 0);
+	m_options.combinedJsonRequests->binary = (requests.count(g_strBinary) > 0);
+	m_options.combinedJsonRequests->binaryRuntime = (requests.count(g_strBinaryRuntime) > 0);
+	m_options.combinedJsonRequests->opcodes = (requests.count(g_strOpcodes) > 0);
+	m_options.combinedJsonRequests->asm_ = (requests.count(g_strAsm) > 0);
+	m_options.combinedJsonRequests->storageLayout = (requests.count(g_strStorageLayout) > 0);
+	m_options.combinedJsonRequests->generatedSources = (requests.count(g_strGeneratedSources) > 0);
+	m_options.combinedJsonRequests->generatedSourcesRuntime = (requests.count(g_strGeneratedSourcesRuntime) > 0);
+	m_options.combinedJsonRequests->srcMap = (requests.count(g_strSrcMap) > 0);
+	m_options.combinedJsonRequests->srcMapRuntime = (requests.count(g_strSrcMapRuntime) > 0);
+	m_options.combinedJsonRequests->funDebug = (requests.count(g_strFunDebug) > 0);
+	m_options.combinedJsonRequests->funDebugRuntime = (requests.count(g_strFunDebugRuntime) > 0);
+	m_options.combinedJsonRequests->signatureHashes = (requests.count(g_strSignatureHashes) > 0);
+	m_options.combinedJsonRequests->natspecDev = (requests.count(g_strNatspecDev) > 0);
+	m_options.combinedJsonRequests->natspecUser = (requests.count(g_strNatspecUser) > 0);
+	m_options.combinedJsonRequests->ast = (requests.count(g_strAst) > 0);
+
+	return true;
+}
+
+size_t CommandLineParser::countEnabledOptions(vector<string> const& _optionNames) const
+{
+	size_t count = 0;
+	for (string const& _option: _optionNames)
+		count += m_args.count(_option);
+
+	return count;
```

Comment:
```suggestion
	return accumulate(transform(_optionNames, [&](auto const& x) { return x.count(_option); }), 0);
```

Not sure how much of range-v3 we want to make use of (I think we at least embrace repeating algorithms?). OTOH, this function does not make any use of `this`, so it might as well be an anonymous function on top of the file?

Reply:
I think we want them in places where they either make expressions simpler or can replace boost stuff.

I'd keep this one as is because the version with a loop is more readable. For me these expressions stop being readable as one-liners once you have to put a lambda in them. I'd put it on a separate line but then it's no longer as concise.